controllers/backplaneconfig_controller.go

```diff
@@ -312,11 +322,62 @@ func (r *MultiClusterEngineReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return result, err
 	}
 
+	if upgrade {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
 	r.StatusManager.AddCondition(status.NewCondition(backplanev1.MultiClusterEngineProgressing, metav1.ConditionTrue, status.DeploySuccessReason, "All components deployed"))
 
 	return ctrl.Result{}, nil
 }
 
+// This function set the operator condition created by OLM to either allow or disallow upgrade based on whether X.Y desired version matches current version
+// It returns an error as well as Boolean determining whether or not the reconcile needs to be rerun in order to update status
+
+func (r *MultiClusterEngineReconciler) setOperatorUpgradeableStatus(ctx context.Context, m *backplanev1.MultiClusterEngine) (bool, error) {
+
+	// Temporary variable
+	var upgradeable bool
+
+	// Checking to see if the current version of the MCE matches the desired to determine if we are in an upgrade scenario
+	// If the current version doesn't exist, we are currently in a install which will also not allow it to upgrade
+	parts1 := strings.Split(m.Status.CurrentVersion, ".")
+	parts2 := strings.Split(version.Version, ".")
+
+	if parts1[0] == parts2[0] && parts1[1] == parts2[1] {
+		upgradeable = true
+	} else {
+		upgradeable = false
+	}
```

Comment:
What should happen if `m.Status.CurrentVersion` is empty? Then `parts1[0]` would equal `""` and `parts1[1]` would equal `panic:indexOutOfBoundsError`. 

Some unit tests would help cover the different scenarios it encounters. I personally think if currentVersion is empty then it should be upgradable because we know we aren't in a y-stream upgrade scenario.

Reply:
@JakobGray When we discussed it for MCE we had said that we do not want to allow upgrade during install in which case `CurrentVersion` is empty. That is how it is working currently and does not panic in this case

Reply:
You're right, we initially said we'll block during install. I think it might be worth revisiting later as this could potentially lock up a z-stream upgrade scenario.